tests/greendns_test.py

```diff
@@ -854,3 +854,15 @@ def test_hosts_priority():
 def test_import_rdtypes_then_eventlet():
     # https://github.com/eventlet/eventlet/issues/479
     tests.run_isolated('greendns_import_rdtypes_then_eventlet.py')
+
+
+def test_nonameserver_is_risen():
```

Comment:
These apply to all code:

- test is named `nonameserver_is_risen` but it doesn't check for any exception
- imports inside function is a recipe for disaster, use it only as last resort
- `reset_default_resolver()` modifies global state, so it renders all other tests results unreliable

This is eventlet project specific, some don't mind: don't rely on external services. Maybe you can't access 8.8.8.8, or this address is not what you expect or it doesn't know about `github.com`.

Solutions:
- for naming - name should be consistent with code
- for imports and global state - isolated tests, see code above
- for external dependencies - mock network IO or run small DNS server that's programmed to respond what you want